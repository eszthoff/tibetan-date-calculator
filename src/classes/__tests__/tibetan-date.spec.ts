import MockDate from 'mockdate';
import TibetanDate from '../tibetan-date';

describe('TibetanDate', () => {
  const mockToday = '2019-06-07';
  let tDate;

  beforeEach(() => {
    MockDate.set(`${mockToday} 12:00`);
    tDate = new TibetanDate();
  });
  afterEach(() => {
    MockDate.reset();
  });

  it('should create a correct class when invoked without arg', () => {
    expect(new TibetanDate()).toMatchSnapshot();
  });
  it('should create a correct class when invoked with string arg', () => {
    expect(new TibetanDate('2019/06/22 12:00')).toMatchSnapshot();
  });
  it('should create a correct class when invoked with object arg', () => {
    expect(new TibetanDate({
      year: 2146, month: 3, day: 10, isLeapMonth: false, isLeapDay: false
    })).toMatchSnapshot();
  });
  it('should get day correctly', () => {
    expect(tDate.day).toEqual(5);
  });
  it('should return year object correctly', () => {
    expect(tDate.yearObj).toMatchSnapshot();
  });
  it('should get western date string correctly', () => {
    expect(tDate.westernDateStr).toEqual(mockToday);
  });
  describe('JS Date like getters', () => {
    it('should return getWesternDate correctly', () => {
      expect(tDate.getWesternDate()).toEqual(tDate.westernDate);
    });
    it('should return getDate correctly', () => {
      expect(tDate.getDate()).toEqual(tDate.date);
    });
    it('should return getDay correctly', () => {
      expect(tDate.getDay()).toEqual(tDate.day);
    });
    it('should return getMonth correctly', () => {
      expect(tDate.getMonth()).toEqual(tDate.month);
    });
    it('should return getMonthObj correctly', () => {
      expect(tDate.getMonthObj()).toEqual(tDate.monthObj);
    });
    it('should return getYear correctly', () => {
      expect(tDate.getYear()).toEqual(tDate.year);
    });
    it('should return getYearObj correctly', () => {
      expect(tDate.getYearObj()).toEqual(tDate.yearObj);
    });
  });
  it('should return the correct string when `toString` is called in not double day', () => {
    expect(new TibetanDate().toString()).toEqual('2146-4-4');
  });
  it('should return the correct string when `toString` is called in leap of double month', () => {
    expect(new TibetanDate('2020/04/24').toString()).toEqual('2147-3-2-leap');
  });
  it('should return the correct string when `toString` is called in main of double month', () => {
    expect(new TibetanDate('2020/04/25').toString()).toEqual('2147-3-2-main');
  });
});
