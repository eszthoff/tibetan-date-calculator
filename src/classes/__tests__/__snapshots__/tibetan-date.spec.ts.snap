// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TibetanDate should create a correct class when invoked with object arg 1`] = `
TibetanDate {
  "date": 10,
  "isDoubledDay": false,
  "isLeapDay": false,
  "isPreviousSkipped": false,
  "isSkippedDay": false,
  "month": 3,
  "monthObj": TibetanMonth {
    "days": Array [],
    "endDateStr": "2019-06-03",
    "isDoubledMonth": false,
    "isLeapMonth": false,
    "month": 3,
    "startDateStr": "2019-05-05",
    "year": 2146,
  },
  "westernDate": 2019-05-14T10:00:00.000Z,
  "year": 2146,
}
`;

exports[`TibetanDate should create a correct class when invoked with string arg 1`] = `
TibetanDate {
  "date": 20,
  "isDoubledDay": false,
  "isLeapDay": false,
  "isPreviousSkipped": false,
  "isSkippedDay": false,
  "month": 4,
  "monthObj": TibetanMonth {
    "days": Array [],
    "endDateStr": "2019-07-02",
    "isDoubledMonth": false,
    "isLeapMonth": false,
    "month": 4,
    "startDateStr": "2019-06-04",
    "year": 2146,
  },
  "westernDate": 2019-06-22T10:00:00.000Z,
  "year": 2146,
}
`;

exports[`TibetanDate should create a correct class when invoked without arg 1`] = `
TibetanDate {
  "date": 4,
  "isDoubledDay": false,
  "isLeapDay": false,
  "isPreviousSkipped": false,
  "isSkippedDay": false,
  "month": 4,
  "monthObj": TibetanMonth {
    "days": Array [],
    "endDateStr": "2019-07-02",
    "isDoubledMonth": false,
    "isLeapMonth": false,
    "month": 4,
    "startDateStr": "2019-06-04",
    "year": 2146,
  },
  "westernDate": 2019-06-07T10:00:00.000Z,
  "year": 2146,
}
`;

exports[`TibetanDate should return year object correctly 1`] = `
TibetanYear {
  "animal": "Pig",
  "element": "Earth",
  "gender": "Female",
  "months": Array [],
  "rabjungCycle": 17,
  "rabjungYear": 33,
  "tibYearNum": 2146,
  "westernYear": 2019,
}
`;
